kitsune/coolsearch/api.py

```diff
@@ -0,0 +1,249 @@
+import json
+from itertools import chain
+
+from django.conf import settings
+from django.http import HttpResponse
+from django.utils.html import escape
+from django.views.generic import View
+
+import bleach
+import jinja2
+from elasticutils.utils import format_explanation
+from tower import ugettext as _, ugettext_lazy as _lazy
+
+from kitsune.coolsearch.forms import SearchForm
+from kitsune.forums.models import ThreadMappingType
+from kitsune.questions.models import QuestionMappingType
+from kitsune.search import es_utils
+from kitsune.search.utils import locale_or_default, clean_excerpt, ComposedList
+from kitsune.sumo.utils import smart_int
+from kitsune.wiki.models import DocumentMappingType
+
+
+EXCERPT_JOINER = _lazy(u'...', 'between search excerpts')
+
+
+class SearchView(View):
+    content_type = 'application/json'
+
+    def get(self, request):
+        search_form = self.form_class(request.GET)
+        if not search_form.is_valid():
+            return HttpResponse(
+                json.dumps({'error': _('Invalid search data.')}),
+                content_type=self.content_type,
+                status=400,
+            )
+
+        language = locale_or_default(
+            request.GET.get('language', request.LANGUAGE_CODE)
+        )
+        lang = language.lower()
+        if settings.LANGUAGES_DICT.get(lang):
+            lang_name = settings.LANGUAGES_DICT[lang]
+        else:
+            lang_name = ''
+
+        page = max(smart_int(request.GET.get('page')), 1)
```

Comment:
I've had a lot of luck using [Django Rest Framework serializers](https://github.com/tomchristie/django-rest-framework/blob/e11f41ebc4ef088a5849771dfda5a7fba4f82904/docs/api-guide/serializers.md#serializers) for parsing parameters like this.